libvirt/tests/src/virtual_network/qemu/bridge_qinq.py

```
@@ -0,0 +1,357 @@
+    session = vm.wait_for_login(timeout=login_timeout)
+    guest_qinq_dir = params["guest_qinq_dir"]
+    copy_qinq_file(vm, guest_qinq_dir)
+    session.close()
+    vm.destroy(gracefully=True)
+
+    # Create libvirt network instead of direct bridge
+    network_name = params.get("network_name", "qinq_test_net")
+    bridge_name = params.get("private_bridge", "tmpbr")
+
+    # Create network using NetworkXML
+    network_dict = eval(params.get("network_dict"))
```

Ruff (0.13.1)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)